src/psyclone/psyir/transformations/psy_data_trans.py

```diff
@@ -0,0 +1,192 @@
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors J. Henrichs, Bureau of Meteorology
+
+'''Contains the PSyData transformation.
+'''
+
+from psyclone.psyGen import Node, Schedule
+from psyclone.psyir.transformations.region_trans import RegionTrans
+from psyclone.psyir.transformations.transformation_error \
+    import TransformationError
+from psyclone.undoredo import Memento
+
+
+class PSyDataTrans(RegionTrans):
+    ''' Create a PSyData region around a list of statements. For
+    example:
+
+    >>> from psyclone.parse.algorithm import parse
+    >>> from psyclone.parse.utils import ParseError
+    >>> from psyclone.psyGen import PSyFactory, GenerationError
+    >>> api = "gocean1.0"
+    >>> filename = "nemolite2d_alg.f90"
+    >>> ast, invokeInfo = parse(filename, api=api, invoke_name="invoke")
+    >>> psy = PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.psyGen import TransInfo
+    >>> t = TransInfo()
+    >>> data_trans = t.get_trans_name('PSyDataTrans')
+    >>>
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    >>> # Enclose all children within a single profile region
+    >>> newschedule, _ = data_trans.apply(schedule.children)
+    >>> newschedule.view()
+
+    '''
+    # Unlike other transformations we can be fairly relaxed about the nodes
+    # that a region can contain as we don't have to understand them.
+    # TODO: #415 Support different classes of PSyData calls.
+    valid_node_types = (Node,)
+
+    def __str__(self):
+        return "Insert a PSyData node."
+
+    @property
+    def name(self):
+        ''' Returns the name of this transformation as a string '''
+        return "PSyDataTrans"
+
+    def validate(self, node_list, options=None):
+        '''
+        Calls the validate method of the base class and then checks that,
+        for the NEMO API, the routine that will contain the profiling
```

Comment:
Please add a TODO for #435 here so we are more likely to remember to update this docstring when #435 finally happens. Oh, and it's not a profiling region anymore.

Reply:
Done.